serializer/serializer.go

```diff
@@ -0,0 +1,28 @@
+package serializer
+
+import (
+	"encoding/json"
+	log "github.com/Sirupsen/logrus"
+)
+
+type Report struct {
+	HttpRequest  string `json:"request"`
+	HttpResponse string `json:"respond"`
+	Source       string `json:"source"`
+	Destination  string `json:"destination"`
+	Description  string `json:"description"`
+}
+
+func UnmarshalReport(data []byte, r *Report) error {
+	defer func() {
```

Comment:
A different way might be to do something like this:

``` go
// Acts as a constructor for a new report
func NewReport(data []byte) (*Report, error) {
    r := &Report{}
    // put your defered panic catcher here
    if err := json.Unmarshal(data, r); err != nil {
        return nil, err
    }

    return r, nil
}
```

Then other parts of the apps can just call `r, err = serializer.NewReport(data)`

You could then hang of a `serialize` function off the `Report` type that serialized the report to json:

``` go
func (r *Report) ToJSON()([]byte, err) {
     return json.Marshal(r)
}
```